### PR TITLE
Add params type checking for mail helper methods

### LINKED ISSUE #701: Type Exception Handling for the Mail Helper
#### Issue Summary

The appropriate types necessary for a successful `mail/send` API call is documented [here](https://sendgrid.com/docs/API_Reference/api_v3.html). We want to throw a type exception when the wrong type is provided.

#### Acceptance Criteria

* All parameters for the `mail\send` endpoint are type checked and if they fail the type check and exception is thrown with instructions on how to fix

### PR DESCRIPTION
**Pull request by f1x3d**
# Resolves #701 

### Checklist
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guide] and my PR follows them.
- [x] I updated my branch with the master branch.
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation about the functionality in the appropriate .md file
- [x] I have added in line documentation to the code I modified

### Short description of what this PR does:
- Add the decorator for type checking of method parameters for mail helper
- Add a test that verifies the implemented functionality

Since the test coverage was not a part of the issue's acceptance criteria, I've included only one basic test.
Please let me know if you want me to add more tests to cover all methods from the mail helper, which are being type checked now.

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/sendgrid-python?pullRequest=720) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/sendgrid-python?pullRequest=720) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/sendgrid-python?pullRequest=720) it.</sub>

**Comment by thinkingserious** [comment]
Hello @f1x3d,<br /><br/>Thanks again for the PR!<br />                                              <br />It's <a href="https://sendgrid.com/blog/hacktoberfest-2018-has-arrived/">HACKTOBERFEST</a>! We want to show our appreciation by sending                                              you some special Hacktoberfest swag. If you have not already, could you please fill out                                              <a href="https://dx.sendgrid.com/hacktoberfest/join">this form</a> so we can send it to you? Thanks!                                             <br /><br />Team SendGrid DX

**Comment by thinkingserious** [comment]
Hello @f1x3d,<br /><br/>Thanks again for the PR!<br />                                              <br />It's <a href="https://sendgrid.com/blog/hacktoberfest-2018-has-arrived/">HACKTOBERFEST</a>! We want to show our appreciation by sending                                              you some special Hacktoberfest swag. If you have not already, could you please fill out                                              <a href="https://dx.sendgrid.com/hacktoberfest/join">this form</a> so we can send it to you? Thanks!                                             <br /><br />Team SendGrid DX

**Comment by f1x3d** [comment]
Hello @thinkingserious,

I've filled the form.

Thank you so much for joining the Hacktoberfest event and rewarding contributors with some awesome swag!

File: .travis.yml [hunk 1]
```diff
@@ -19,6 +19,7 @@ install:
 - pip install pypandoc
 - pip install coverage
 - pip install codecov
+- pip install wrapt
```
Comment (eshanholtz) [inline .travis.yml:22]: Why add a new dependency instead of using built-in decorators?

File: sendgrid/helpers/mail/decorators.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    def wrapper(wrapped, instance, args, kwargs):
+        args_names = inspect.getargspec(wrapped)[0]
+        if instance is not None:
+            args_names = args_names[1:]
+
+        for (arg_val, expected_type, arg_name) in zip(args, types, args_names):
+            if expected_type is str:
+                expected_type = six.string_types
+
+            if not isinstance(arg_val, expected_type):
```
Comment (eshanholtz) [inline sendgrid/helpers/mail/decorators.py:20]: I'm worried that this is going to cause more breaking changes than is functionally necessary. For example, with [this PR](https://github.com/sendgrid/sendgrid-python/pull/920) adding strict type checking broke a functional implementation.

**Comment by childish-sambino** [comment]
Closing until PR comments have been addressed.

**Pull request closed without merging**